services/121-service/src/program-financial-service-provider-configurations/program-financial-service-provider-configurations.service.ts

```diff
@@ -1,123 +1,448 @@
+import { ProgramFinancialServiceProviderConfigurationResponseDto } from '@121-service/src/program-financial-service-provider-configurations/dtos/program-financial-service-provider-configuration-response.dto';
+import { UpdateProgramFinancialServiceProviderConfigurationDto } from '@121-service/src/program-financial-service-provider-configurations/dtos/update-program-financial-service-provider-configuration.dto';
+import { UpdateProgramFinancialServiceProviderConfigurationPropertyDto } from '@121-service/src/program-financial-service-provider-configurations/dtos/update-program-financial-service-provider-configuration-property.dto';
+import { ProgramFinancialServiceProviderConfigurationEntity } from '@121-service/src/program-financial-service-provider-configurations/entities/program-financial-service-provider-configuration.entity';
+import { ProgramFinancialServiceProviderConfigurationPropertyEntity } from '@121-service/src/program-financial-service-provider-configurations/entities/program-financial-service-provider-configuration-property.entity';
+import { ProgramFinancialServiceProviderConfigurationMapper } from '@121-service/src/program-financial-service-provider-configurations/mappers/program-financial-service-provider-configuration.mapper';
+import { ProgramEntity } from '@121-service/src/programs/program.entity';
+import { ProgramRepository } from '@121-service/src/programs/repositories/program.repository';
 
 @Injectable()
 export class ProgramFinancialServiceProviderConfigurationsService {
   @InjectRepository(ProgramFinancialServiceProviderConfigurationEntity)
   private readonly programFspConfigurationRepository: Repository<ProgramFinancialServiceProviderConfigurationEntity>;
+  @InjectRepository(ProgramFinancialServiceProviderConfigurationPropertyEntity)
+  private readonly programFspConfigurationPropertyRepository: Repository<ProgramFinancialServiceProviderConfigurationPropertyEntity>;
+  @InjectRepository(ProgramEntity)
+  private readonly programRepository: ProgramRepository;
 
   public async findByProgramId(
     programId: number,
-  ): Promise<ProgramFinancialServiceProviderConfigurationEntity[]> {
+  ): Promise<ProgramFinancialServiceProviderConfigurationResponseDto[]> {
+    await this.validateProgramExists(programId);
+
     const programFspConfigurations =
       await this.programFspConfigurationRepository.find({
         where: { programId: Equal(programId) },
+        relations: ['properties'],
+      });
+
+    return ProgramFinancialServiceProviderConfigurationMapper.mapEntitiesToDtos(
+      programFspConfigurations,
+    );
+  }
+
+  public async validateAndCreate(
+    programId: number,
+    programFspConfigurationDto: CreateProgramFinancialServiceProviderConfigurationDto,
+  ): Promise<ProgramFinancialServiceProviderConfigurationResponseDto> {
+    await this.validate(programId, programFspConfigurationDto);
+    return this.create(programId, programFspConfigurationDto);
+  }
+
+  public async validate(
```

Comment:
this method is public only for the purpose of the unit test, which shouldn't be the case right? Instead, should we unit test validateAndCreate, which is rightfully public and is missing a unit test now?

Reply:
Good point. At first I also used validate in the program service which i later on deleted.